src/fontra/views/editor/panel-transformation.js

```diff
@@ -450,11 +584,341 @@ export default class TransformationPanel extends Panel {
+      sceneController.experimentalFeatures.scalingEditBehavior
+    );
+
+    const editBehavior = behaviorFactory.getBehavior("default");
+    const editChange = editBehavior.makeChangeForDelta(delta);
+    return [editChange, editBehavior.rollbackChange];
+  }
+}
+
+class MovablePoint extends MovableBaseObject {
+  constructor(pointIndex, selection) {
+    super(selection);
+    this.pointIndex = pointIndex;
+  }
+}
+
+class MovableContour extends MovableBaseObject {
+  constructor(pointIndices, selection) {
+    super(selection);
+    this.pointIndices = pointIndices;
+  }
+}
+
+class MovableComponent extends MovableBaseObject {
+  constructor(componentIndex, selection) {
+    super(selection);
+    this.componentIndex = componentIndex;
+  }
+}
+
+// Define moveDescriptor objects
+const alignLeft = {
+  undoLabel: "align left",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const xMins = boundingBoxes.map((bounds) => bounds.xMin);
+    const left = Math.min(...xMins);
+    return xMins.map((xMin) => ({
+      x: left - xMin,
+      y: 0,
+    }));
+  },
+};
+
+const alignCenter = {
+  undoLabel: "align center",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const xMaxes = boundingBoxes.map((bounds) => bounds.xMax);
+    const xMins = boundingBoxes.map((bounds) => bounds.xMin);
+    const left = Math.min(...xMins);
+    const right = Math.max(...xMaxes);
+    return boundingBoxes.map((bounds) => ({
+      x: left - bounds.xMin + (right - left) / 2 - (bounds.xMax - bounds.xMin) / 2,
+      y: 0,
+    }));
+  },
+};
+
+const alignRight = {
+  undoLabel: "align right",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const xMaxes = boundingBoxes.map((bounds) => bounds.xMax);
+    const right = Math.max(...xMaxes);
+    return xMaxes.map((xMax) => ({
+      x: right - xMax,
+      y: 0,
+    }));
+  },
+};
+
+const alignTop = {
+  undoLabel: "align top",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const yMaxes = boundingBoxes.map((bounds) => bounds.yMax);
+    const top = Math.max(...yMaxes);
+    return yMaxes.map((yMax) => ({
+      x: 0,
+      y: top - yMax,
+    }));
+  },
+};
+
+const alignMiddle = {
+  undoLabel: "align middle",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const yMaxes = boundingBoxes.map((bounds) => bounds.yMax);
+    const yMins = boundingBoxes.map((bounds) => bounds.yMin);
+    const bottom = Math.min(...yMins);
+    const top = Math.max(...yMaxes);
+    return boundingBoxes.map((bounds) => ({
+      x: 0,
+      y: top - bounds.yMax + (bounds.yMax - bounds.yMin) / 2 - (top - bottom) / 2,
+    }));
+  },
+};
+
+const alignBottom = {
+  undoLabel: "align bottom",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const yMins = boundingBoxes.map((bounds) => bounds.yMin);
+    const bottom = Math.min(...yMins);
+    return yMins.map((yMin) => ({
+      x: 0,
+      y: bottom - yMin,
+    }));
+  },
+};
+
+const distributeHorizontally = {
```

Comment:
`distributeHorizontally` and `distributeVertically` are identical apart from variable names and whether they use xMin/xMax or yMin/yMax. We can refactor as follows.

```js
class DistributeObjectsDescriptor {

  constructor(direction, minProperty, maxProperty) {
    this.undoLabel = `distribute ${direction}`;
    this.minProperty = minProperty;
    this.maxProperty = maxProperty;
  }

  computeDeltasFromBoundingBoxes(boundingBoxes) {
    // original function content, but writing
    //   `bounds[this.minProperty]`
    // instead of bounds.xMin, and
    //   `bounds[this.maxProperty]`
    // instead of bounds.xMax
    // And use `minimum` instead of `left` and `maximum` instead of `right`
  }

  sortObjects(a, b, controller) {
    return a.computeBounds(controller)[this.minProperty] - b.computeBounds(controller)[this.minProperty]
    // Btw. I still think this should sort based on the center, not the left or bottom
  }
}

const distributeHorizontally = DistributeObjectsDescriptor("horizontally", "xMin", "xMax");
const distributeVertically = DistributeObjectsDescriptor("vertically", "yMin", "yMax");
```

Reply:
Done.